File: AvaloniaUI/HiringDashboard/Views/SidebarButton.axaml.cs
using Avalonia;
using Avalonia.Controls;
using Avalonia.Markup.Xaml;
using Avalonia.Media;

namespace HiringDashboard.Controls
{
    public partial class SidebarButton : Button
    {
        public static readonly AttachedProperty<Geometry> IconDataProperty =
            AvaloniaProperty.RegisterAttached<SidebarButton, Button, Geometry>("IconData");

        public static void SetIconData(Button element, Geometry value)
        {
            element.SetValue(IconDataProperty, value);
        }

        public static Geometry GetIconData(Button element)
        {
            return element.GetValue(IconDataProperty);
        }

        public static readonly AttachedProperty<string> ButtonTextProperty =
            AvaloniaProperty.RegisterAttached<SidebarButton, Button, string>("ButtonText");

        public static void SetButtonText(Button element, string value)
        {
            element.SetValue(ButtonTextProperty, value);
        }

        public static string GetButtonText(Button element)
        {
            return element.GetValue(ButtonTextProperty);
        }

        private void InitializeComponent()
        {
            AvaloniaXamlLoader.Load(this);
        }

        public SidebarButton()
        {
            InitializeComponent();
        }
    }
}

File: AvaloniaUI/HiringDashboard/Views/SidebarButton.axaml.fs
namespace HiringDashboard.Controls

open Avalonia
open Avalonia.Controls
open Avalonia.Markup.Xaml
open Avalonia.Media

type SidebarButton() as this =
    inherit Button()
    
    static let IconDataProperty =
        AvaloniaProperty.Register<SidebarButton, string>("IconData", "")
        
    static let ButtonTextProperty =
        AvaloniaProperty.Register<SidebarButton, string>("ButtonText", "")
        
    static let IsCheckedProperty =
        AvaloniaProperty.Register<SidebarButton, bool>("IsChecked", false)
    
    do AvaloniaXamlLoader.Load(this)
    
    member this.IconData
        with get() = this.GetValue(IconDataProperty)
        and set(value) = 
            this.SetValue(IconDataProperty, value) |> ignore
    
    member this.ButtonText
        with get() = this.GetValue(ButtonTextProperty)
        and set(value) = 
            this.SetValue(ButtonTextProperty, value) |> ignore
            
    member this.IsChecked
        with get() = this.GetValue(IsCheckedProperty)
        and set(value) = 
            this.SetValue(IsCheckedProperty, value) |> ignore 